### ISSUE TITLE
Intermittent github 404

### ISSUE THREAD

**Issue by jazanne** [lycheeverse/lychee#431]
These 2 links give me intermittent 404s when I run Lychee GitHub action.

I have lots of other github links that never 404, it's only these two. Any thought on why that could be?

My config

```yaml
linkChecker:
    name: "Check external links"
    runs-on: ubuntu-latest
    steps:
      - uses: actions/checkout@v2
      - name: Link Checker
        uses: lycheeverse/lychee-action@v1.2.0
        with:
          args: --config .github/lychee.toml **/*.mdx
          fail: true
        env:
          GITHUB_TOKEN: ${{secrets.GITHUB_TOKEN}}
```

* [https://github.com/features/actions](https://github.com/features/actions): Failed: 404 Not Found: Not Found
* [https://github.com/marketplace/actions/launchdarkly-code-references](https://github.com/marketplace/actions/launchdarkly-code-references): Failed: 404 Not Found: Not Found

**Comment by mre** [comment]
Huh? That works for me. 🤔 

`foo.md`:
```
https://github.com/features/actions
https://github.com/marketplace/actions/launchdarkly-code-references
```

```
> lychee /tmp/foo.md -v                                                                                                                         ✘
✔ https://github.com/features/actions [200 OK]
✔ https://github.com/marketplace/actions/launchdarkly-code-references [200 OK]

🔍 2 Total ✅ 2 OK 🚫 0 Errors
```

**Comment by mre** [comment]
Anything special in the `lychee.toml`?

**Comment by jazanne** [comment]
not particularly - here's my config

```toml
###
### Display
###
# Verbose program output
verbose = true

# Show progress
progress = false

###
### Exclusions
###
# Exclude URLs from checking (supports regex)
exclude = [
# some private links in here
]

# Exclude all private IPs from checking
# Equivalent to setting `exclude_private`, `exclude_link_local`, and `exclude_loopback` to true
exclude_all_private = true
exclude_mail = true
```

**Comment by mre** [comment]
Which lychee version do you use? The latest from `master` or a different one? The current version in the lychee-action is still [0.8.1](https://github.com/lycheeverse/lychee-action/blob/f76b8412c668f78311212d16d33c4784a7d8762c/Dockerfile#L1) (not that it should make a difference).

**Comment by jazanne** [comment]
0.8.1

**Comment by mre** [comment]
Tried it with 0.8.1 and with the lychee-action. In both cases it worked for me.
Do the links only fail occasionally or consistently for you?
Is the code public somewhere so that I can have a look?
Is the link in a Markdown file or in an HTML file?

**Comment by mre** [comment]
Come to think of it, I got 404s before with other Github URLs like sponsor pages, e.g.
https://github.com/sponsors/analysis-tools-dev.
Could be related and we might need another quirk: https://github.com/lycheeverse/lychee/blob/master/lychee-lib/src/quirks/mod.rs

**Comment by mre** [comment]
I think one reason is that our Github extractor is too permissive:
https://github.com/lycheeverse/lychee/blob/166c86c30ee3c93dbdfc9f8948f402c969e487df/lychee-lib/src/types/uri.rs#L70-L85

It doesn't handle special cases like https://github.com/sponsors/analysis-tools-dev and will happily extract this as an (org, repo) tuple and check it through the Github API, which leads to an error because it's not a valid project URL. So the underlying issue might be that you had too many requests to Github, which triggered a 429 and then we fall-back on the Github API, which cannot find the repository and returns a 404.

To fix this, we need to return `None` from the github extractor if the path begins with any of the following:
* sponsors
* features
* marketplace

There might be other special Github endpoints, which should excluded as well. Happy for suggestions.

**Comment by mre** [comment]
@jazanne, can you try again with the latest `master`? I merged #445, which should fix the false-positives you mentioned. It might still fail if you have a lot of Github links, but with a different status code (429 instead of 404).

**Comment by mre** [comment]
If that's not enough, you can also reduce the number of retries (currently 3 by default) after https://github.com/lycheeverse/lychee/pull/446 is merged. Usage example: `--max-retries 0`.
